scripts/globals/utils.lua

```diff
@@ -394,3 +394,62 @@ function utils.hasRelic(player, relic, tier)
 
     return false
 end
+
+-- utils.mask contains functions for bitmask variables
+utils.mask =
+{
+    -- return mask's pos-th bit as bool
+    getBit = function(mask, pos)
+        return bit.band(mask, bit.lshift(1, pos)) ~= 0
+    end,
+
+    -- return value of mask after setting its pos-th bit
+    -- val can be bool or number.  if number, any non-zero value will be treated as true.
+    setBit = function(mask, pos, val)
+        local state = false
+
+        if type(val) == "boolean" then
+            state = val
+        elseif type(val) == "number" then
+            state = (val ~= 0)
+        end
+
+        if state then
+            -- turn bit on
+            return bit.bor(mask, bit.lshift(1, pos))
+        else
+            -- turn bit off
+            return bit.band(mask, bit.bnot(bit.lshift(1, pos)))
+        end
+    end,
+
+    -- return number of true bits in mask of length len
+    -- if len is omitted, assume 32
+    countBits = function(mask, len)
+        if not len then
+            len = 32
+        end
+
+        local count = 0
+
+        for i = 0, len - 1 do
+            if bit.band(mask, bit.lshift(1, i)) ~= 0 then
+                count = count + 1
+            end
+        end
```

Comment:
Cool beans!

I'm sure this doesn't *have* to be micro-optimized, but simply adding all the bits without the `if` would be a bit neater.
```lua
count = count + bit.band(bit.rshift(mask, i), 1)
```

Reply:
I like it!  Implemented, retested, and looks good.